### ISSUE TITLE
Document high contrast mode for accessibility

### ISSUE THREAD

**Issue by loic-sharma** [flutter/website#9769]
### What information needs to be added?

Apps should support high contrast mode for accessibility. Currently Flutter doesn't have any documentation on how to support this properly. We should explain what high contrast mode is, which platforms support it, how to test it, etc..

### Where should this new content appear?

Accessibility & Internalization section

### I would like to fix this problem.

- [ ] I will try and fix this problem on docs.flutter.dev.

**Comment by atsansone** [comment]
@loic-sharma : This makes sense. Do you have any starting point for documentation? Or do we need to meet and review?

**Comment by loic-sharma** [comment]
I'd be happy to meet and review. To the best of my knowledge, Flutter has no documentation for this topic. Here are some resources that might be helpful:

1. Web accessibility guidelines: https://www.w3.org/WAI/WCAG22/quickref/?versions=2.1&showtechniques=146#contrast-enhanced
2. Windows docs: https://learn.microsoft.com/en-us/windows/win32/w8cookbook/high-contrast-mode
3. Apple docs: https://developer.apple.com/documentation/accessibility/testing-system-accessibility-features-in-your-app#Provide-sufficient-color-contrast
4. @goderbauer's investigations of high contrast with Material: https://github.com/flutter/flutter/issues/75883#issuecomment-881697023
